pkg/utils/kubernetes/highavailability.go

```diff
@@ -40,8 +40,8 @@ func GetReplicaCount(failureToleranceType *gardencorev1beta1.FailureToleranceTyp
 
 // GetNodeAffinitySelectorTermsForZones adds a node affinity to ensure all pods are scheduled only on nodes in the provided zones. If
 // no zones are provided then nothing is done.
-func GetNodeAffinitySelectorTermsForZones(failureToleranceType *gardencorev1beta1.FailureToleranceType, zones []string) []corev1.NodeSelectorTerm {
-	if len(zones) == 0 || failureToleranceType == nil {
+func GetNodeAffinitySelectorTermsForZones(isZonePinningEnabled bool, zones []string) []corev1.NodeSelectorTerm {
```

Comment:
Adapt unit and integration tests

Reply:
Adapted unit tests in `highavailability_test.go`.

Which integration test did you have in mind?

Reply:
Found and adapted it.